packages/@uppy/dashboard/src/Dashboard.jsx

```diff
@@ -67,7 +67,13 @@ export default class Dashboard extends UIPlugin {
       hidePauseResumeButton: false,
       hideProgressAfterFinish: false,
       doneButtonHandler: () => {
-        this.uppy.cancelAll()
+        this.uppy.setState({
+          totalProgress: 0,
+          error: null,
+          recoveredState: null,
+          allowNewUpload: true,
+          files: {},
+        })
```

Comment:
Note that this is practically identical to `resetProgress` with the only difference that `files` aren't removed. Should indeed be streamlined.

https://github.com/transloadit/uppy/blob/12e08ada02b9080bd5e1d19526bdf8a2010e62a1/packages/%40uppy/core/src/Uppy.js#L233-L261

Reply:
yes i also suggested that during the call, but the reasoning was that it's very slow because it iterates over all files. but in this case we don't need to do that because all files will be removed anyways. but yea we should probably reuse parts of that function

Reply:
pushed some code reuse